synapse/storage/persist_events.py

```diff
@@ -660,3 +681,55 @@ def _maybe_start_persisting(self, room_id: str):
         }
 
         return DeltaState(to_delete=to_delete, to_insert=to_insert)
+
+    async def _is_server_still_joined(
+        self,
+        room_id: str,
+        ev_ctx_rm: List[Tuple[FrozenEvent, EventContext]],
+        delta: DeltaState,
+        current_state: Optional[StateMap[str]],
+    ) -> bool:
+        # Now we check if we're actually still in the room,
```

Comment:
this comment probably needs turning into a docstring

Reply:
You know, I looked at it and thought it was a docstring :man_facepalming: